sycl/include/CL/sycl/xilinx/fpga/partition_array.hpp

```diff
@@ -52,7 +52,10 @@ namespace partition {
   /// This fuction is currently empty but the LowerSYCLMetaData Pass will fill
   /// it with the required IR.
   template<typename Ptr>
-  __SYCL_DEVICE_ANNOTATE("xilinx_partition_array") __attribute__((always_inline))
+#if defined(__SYCL_XILINX_HW_EMU_MODE__) || defined(__SYCL_XILINX_HW_MODE__)
+  __SYCL_DEVICE_ANNOTATE("xilinx_partition_array")
+#endif
+  __attribute__((always_inline))
```

Comment:
Would this compile with MSVC?
It would be nice to have this work in plain CPU mode.

Reply:
> Would this compile with MSVC?
I don't think so because the __attribute__ attribute format isn't supported by msvc as far AFAIK.
> It would be nice to have this work in plain CPU mode.
yes on gcc/clang because __SYCL_DEVICE_ANNOTATE will disappear.

Reply:
Using a macro like `__SYCL_ALWAYS_INLINE` or `__TRISYCL_ALWAYS_INLINE`?

Reply:
ALWAYS_INLINE is the macro intel added so even if we create a new one we are not going to be shielded from collision on there code.